src/coprocessor/dag/expr/ctx.rs

```diff
@@ -46,6 +20,71 @@ bitflags! {
     }
 }
 
+bitflags! {
+    /// Flags are used by `DAGRequest.flags` to handle execution mode, like how to handle
+    /// truncate error.
+    pub struct Flags: u64 {
+        /// `FLAG_IGNORE_TRUNCATE` indicates if truncate error should be ignored.
+        /// Read-only statements should ignore truncate error, write statements should not ignore
+        /// truncate error.
+        const FLAG_IGNORE_TRUNCATE = 1;
+        /// `FLAG_TRUNCATE_AS_WARNING` indicates if truncate error should be returned as warning.
+        /// This flag only matters if `FLAG_IGNORE_TRUNCATE` is not set, in strict sql mode, truncate error
+        /// should be returned as error, in non-strict sql mode, truncate error should be saved as warning.
+        const FLAG_TRUNCATE_AS_WARNING = 1 << 1;
+        // `FLAG_PAD_CHAR_TO_FULL_LENGTH` indicates if sql_mode 'PAD_CHAR_TO_FULL_LENGTH' is set.
+        const FLAG_PAD_CHAR_TO_FULL_LENGTH = 1 << 2;
+        /// `FLAG_IN_INSERT_STMT` indicates if this is a INSERT statement.
+        const FLAG_IN_INSERT_STMT = 1 << 3;
+        /// `FLAG_IN_UPDATE_OR_DELETE_STMT` indicates if this is a UPDATE statement or a DELETE statement.
+        const FLAG_IN_UPDATE_OR_DELETE_STMT = 1 << 4;
+        /// `FLAG_IN_SELECT_STMT` indicates if this is a SELECT statement.
+        const FLAG_IN_SELECT_STMT = 1 << 5;
+        /// `FLAG_OVERFLOW_AS_WARNING` indicates if overflow error should be returned as warning.
+        /// In strict sql mode, overflow error should be returned as error,
+        /// in non-strict sql mode, overflow error should be saved as warning.
+        const FLAG_OVERFLOW_AS_WARNING = 1 << 6;
+
+        // FLAG_DIVIDED_BY_ZERO_AS_WARNING indicates if DividedByZero should be returned as warning.
+        const FLAG_DIVIDED_BY_ZERO_AS_WARNING = 1 << 8;
+    }
+}
+
+impl Flags {
+    /// Returns if 'FLAG_IGNORE_TRUNCATE' mode is set.
+    pub fn is_ignore_truncate(self) -> bool {
```

Comment:
This means that we need an additional function for each flag, which is not very DRY. Maybe you can remove these functions and change them to use flag.contains() directly